phases/12-multimodal-ai/01-vision-transformer-patch-tokens/code/main.py

```
@@ -0,0 +1,183 @@
+            patches.append(patch)
+
+    print(f"\npatches ({len(patches)} total, each length {P*P}):")
+    for i, p in enumerate(patches):
+        print(f"  patch {i}: {p}")
+
+    fake_W = [[((i + j) % 5) - 2 for j in range(P * P)] for i in range(4)]
+    embeddings = []
+    for patch in patches:
+        emb = []
+        for row in fake_W:
+            s = sum(r * v for r, v in zip(row, patch))
```

Ruff (0.15.11)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)